app.py

```diff
@@ -249,10 +241,26 @@ def download_manifest(self):
 
         return manifest_io.getvalue().decode('utf-8')
 
+    def audit_log(self, event: str):
+
+        log_data = f'{datetime.now()} | {event} | {request.remote_addr} | {json.dumps(dict(request.headers))}\n'
+
+        audit_file_name = f'{datetime.today().strftime("%y-%m")}.log'
+        audit_file = BytesIO()
+
+        if self.object_exists(audit_file_name, S3_AUDIT_FOLDER):
+            self.client.download_fileobj(self.bucket, f'{S3_AUDIT_FOLDER}{audit_file_name}', audit_file)
+
+        audit_file.seek(0, io.SEEK_END)
+        audit_file.write(log_data.encode())
+        audit_file.seek(0)
+
+        self.client.upload_fileobj(audit_file, self.bucket, f'{S3_AUDIT_FOLDER}{audit_file_name}')
```

Comment:
There is a little bit scary what audit log is not immutable and may be rewritten entirely

Reply:
Added a code snippet to prevent the log to be rewritten https://github.com/tarantool/rocks.tarantool.org/pull/45/files#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959R273